demos/instr.cc

```diff
@@ -19,7 +19,7 @@
 #include "instr.h"
 
 #if defined(__i386__) || defined(__x86_64__) || defined(_M_X64) || \
-    defined(_M_IX86)
+    defined(_M_AMD64) || defined(_M_IX86)
```

Comment:
The number of places we had to change for this strongly suggests to me we should come up with some more semantically meaningful `#define` that helps hide the blowup of arch+compiler macros.

Something like: https://github.com/abseil/abseil-cpp/blob/9ee91d3e430fb33a4590486573792eb0fa146c2d/absl/random/internal/platform.h#L61-L78

Should we do that here or in another PR?

Reply:
Continue on https://github.com/google/safeside/pull/58